### PR TITLE
Feat/add fail on like cli support

### PR DESCRIPTION
**Pull request by aarlaud**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Adds fail-on support like Snyk cli so the return code can be non 0 only if new vulns are upgradable, or patchable, or both.
Also  unskip a test by fixing fixtures mostly.

### CODE REVIEW

File: test/lib/index-module.test.ts [hunk 1]
```diff
@@ -72,51 +71,270 @@ beforeEach(() => {
 });
 
 describe('Test End 2 End - Module', () => {
-  it('Test module mode - no new issue', async () => {
+  // it('Test module mode - no new issue', async () => {
```
Comment (lili2311) [inline test/lib/index-module.test.ts:74]: intentional?

**Review by lili2311 — CHANGES_REQUESTED** [review]

File: test/lib/index-module.test.ts (hunk 1 above)
Comment (aarlaud) [reply test/lib/index-module.test.ts:74]: argh !!

File: src/lib/index.ts [hunk 2]
```diff
@@ -38,9 +40,16 @@ const getDelta = async(snykTestOutput = '', debugMode = false, setPassIfNoBaseli
...
-    const currentOrg: string = argv.currentOrg ? argv.currentOrg: ""
-    const currentProject: string = argv.currentProject ? argv.currentProject: ""
-
+    const currentOrg: string = argv.currentOrg ? argv.currentOrg : ""
+    const currentProject: string = argv.currentProject ? argv.currentProject : ""
+    const failOnFlag: string = argv["fail-on"]? argv["fail-on"] : failOnFlagFromModule.toLowerCase()
+    debug(failOnFlag)
+
+    if(failOnFlag && !["all","upgradable","patchable"].includes(failOnFlag.toLowerCase())){
+      debug("Fail On flag can only have the following values: [all,upgradable,patchable]");
```
[the remark this answers is not in the document]
Comment (aarlaud) [reply src/lib/index.ts:49]: yargs already takes care of that when using it inline.

File: src/lib/snyk/snyk_utils.ts [hunk 3]
```diff
@@ -1,11 +1,37 @@
+import { IssueWithPaths } from "../types";
+import { getDebugModule } from "../utils/utils";
 //import * as Configstore from '@snyk/configstore';
```
[the remark this answers is not in the document]
Comment (aarlaud) [reply src/lib/snyk/snyk_utils.ts:3]: why? it's used below

[the remark this answers is not in the document]
Comment (lili2311) [reply src/lib/snyk/snyk_utils.ts:3]: To clarify, I meant to drop the commented out line
> //import * as Configstore from '@snyk/configstore';

File: src/lib/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lili2311) [reply src/lib/index.ts:49]: I thought so too, so what does this add then?

File: src/lib/index.ts [hunk 4]
```diff
@@ -38,9 +40,16 @@ const getDelta = async(snykTestOutput = '', debugMode = false, setPassIfNoBaseli
...
+      process.exit(2);
+     }
+debug("failon: ",failOnFlag)
```
Comment (lili2311) [inline src/lib/index.ts:52]: missing spaces/formatting?

**Pull request merged**